### PR TITLE
Add logics to records attempts

### PR DESCRIPTION
**Pull request by Jideboris**
Add logics to records attempts

[ROE-1467](https://companieshouse.atlassian.net/browse/ROE-1467)

### Type of change

* [ ] Bug fix
* [x New feature
* [ ] Breaking change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/golang/go/wiki/CodeReviewComments)__

### CODE REVIEW

File: dao/mongo.go [hunk 1]
```diff
@@ -311,23 +311,37 @@ func (m *MongoService) GetPaymentsWithRefundPendingStatus() ([]models.PaymentRes
...
 		ArrayFilters: &refundsFilter,
 		Upsert:       &upsert,
 	}
 
-	patchUpdate := bson.M{
-		"$set": bson.M{
-			"refunds.$[x].status":      "refund-requested",
-			"refunds.$[x].refunded_at": time.Now(),
-		},
+	patchUpdate := bson.M{}
```
Comment (emead) [inline dao/mongo.go:327]: Would be better as:
```suggestion
	var patchUpdate bson.M
```

File: models/refund_db.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ type RefundResourceDB struct {
 	CreatedAt         string `bson:"created_at"`
 	Amount            int    `bson:"amount"`
 	Status            string `bson:"status"`
+	Attempts          int    `bson:"attempts"`
```
Comment (emead) [inline models/refund_db.go:9]: Does this create an empty `attempts` field upon creation of the refund?
`omitempty` could be used here

File: service/refund.go [hunk 3]
```diff
@@ -505,20 +506,21 @@ func checkGovPayAndUpdateRefundStatus(req *http.Request, payments []models.Payme
...
-			if err != nil {
-				log.ErrorR(req, fmt.Errorf("error updating payment Id: [%s] pending refund status ", x.ID))
-				return fmt.Errorf("error updating payment Id: [%s] pending refund status ", x.ID)
-			}
+		isPaid := govPayStatusResponse != nil && govPayStatusResponse.Status == RefundsStatusSuccess
+
+		err = service.DAO.PatchPaymentsWithRefundPendingStatus(x.ID, isPaid, &x)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("error updating payment Id: [%s] pending refund status ", x.ID))
+			return fmt.Errorf("error updating payment Id: [%s] pending refund status ", x.ID)
```
Comment (emead) [inline service/refund.go:521]: The value of `err` should be included in the log and returned error

File: service/refund.go [hunk 4]
```diff
@@ -505,20 +506,21 @@ func checkGovPayAndUpdateRefundStatus(req *http.Request, payments []models.Payme
...
 		}
 
-		govPayStatusResponse, response, err := service.GovPayService.GetRefundStatus(paymentSession, x.Refunds[0].RefundId)
+		govPayStatusResponse, response, err := service.GovPayService.GetRefundStatus(paymentSession, refund.RefundId)
 
 		if err != nil {
-			log.ErrorR(req, fmt.Errorf("error getting payment pending refund status Id: [%s]", x.Refunds[0].RefundId))
-			return fmt.Errorf("error getting payment pending refund status Id: [%s]", x.Refunds[0].RefundId)
+			log.ErrorR(req, fmt.Errorf("error getting payment pending refund status Id: [%s]", refund.RefundId))
+			return fmt.Errorf("error getting payment pending refund status Id: [%s]", refund.RefundId)
```
Comment (emead) [inline service/refund.go:513]: The value of `err` should be included in the log and returned error

Comment (Jideboris) [reply service/refund.go:513]: Fixed.Thanks

File: service/refund.go (hunk 3 above)
Comment (Jideboris) [reply service/refund.go:521]: Fixed.Thanks

File: models/refund_db.go (hunk 2 above)
Comment (Jideboris) [reply models/refund_db.go]: I was expecting it to take 0 being integer if empty.Anyway added with lots of thanks.

File: dao/mongo.go (hunk 1 above)
Comment (Jideboris) [reply dao/mongo.go]: Fixed

**Review by emead — APPROVED** [review]

**Pull request merged**